internal/hcs/system.go

```diff
@@ -541,26 +540,64 @@ func (computeSystem *System) CreateProcess(c interface{}) (_ cow.Process, err er
 	})
 	events := processHcsResult(resultp)
 	if err != nil {
-		return nil, makeSystemError(computeSystem, "CreateProcess", configuration, err, events)
+		return nil, 0, makeSystemError(computeSystem, "CreateProcess", configuration, err, events)
 	}
 
 	logrus.WithFields(computeSystem.logctx).
 		WithField(logfields.ProcessID, processInfo.ProcessId).
 		Debug("HCS ComputeSystem CreateProcess PID")
 
+	return &processInfo, processHandle, nil
+}
+
+// CreateProcessNoStdio launches a new process within the computeSystem. The
+// Stdio handles are not cached on the process struct.
+func (computeSystem *System) CreateProcessNoStdio(c interface{}) (_ cow.Process, err error) {
+	processInfo, processHandle, err := computeSystem.createProcess(c)
+	if err != nil {
+		return nil, err
+	}
 	process := newProcess(processHandle, int(processInfo.ProcessId), computeSystem)
 	defer func() {
 		if err != nil {
 			process.Close()
 		}
 	}()
+
+	// We don't do anything with these handles. Close them so they don't leak.
+	syscall.Close(processInfo.StdInput)
+	syscall.Close(processInfo.StdOutput)
+	syscall.Close(processInfo.StdError)
+
+	if err = process.registerCallback(); err != nil {
+		return nil, makeSystemError(computeSystem, "CreateProcess", "", err, nil)
+	}
+	go process.waitBackground()
+
+	return process, nil
+}
+
+// CreateProcess launches a new process within the computeSystem.
+func (computeSystem *System) CreateProcess(c interface{}) (_ cow.Process, err error) {
+	processInfo, processHandle, err := computeSystem.createProcess(c)
+	if err != nil {
+		return nil, err
+	}
+	process := newProcess(processHandle, int(processInfo.ProcessId), computeSystem)
```

Comment:
Why isn't newProcess called by createProcess?

Reply:
Because we still need access to `processInfo` here (for the stdio handles). We could instead make `createProcess` return a process and processInfo (or a slice of handles), do you think that would be better?

Reply:
Yes, I think so.

Reply:
Okay, updated to return process and process info from `createProcess`.